crates/miden-objects/src/block/signed_block.rs

```diff
@@ -0,0 +1,47 @@
+use miden_core::utils::{
+    ByteReader,
+    ByteWriter,
+    Deserializable,
+    DeserializationError,
+    Serializable,
+};
+
+use crate::block::{BlockBody, BlockHeader};
+
+#[derive(Debug, Clone)]
+pub struct SignedBlock {
+    header: BlockHeader,
+    body: BlockBody,
+}
+
+impl SignedBlock {
+    /// Creates a new [`SignedBlock`] with the given header and body.
+    pub fn new(header: BlockHeader, body: BlockBody) -> Self {
```

Comment:
This seems like a candidate for a `new_unchecked` rename because header and body must be consistent and I think we should document this in the docs and in the name.